okta/userFactor.go

```diff
@@ -0,0 +1,406 @@
+/*
+* Copyright 2018 - Present Okta, Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*      http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// AUTO-GENERATED!  DO NOT EDIT FILE DIRECTLY
+
+package okta
+
+import (
+	"bytes"
+	"fmt"
+	"encoding/json"
+	"time"
+)
+
+type CallFactor struct {
+	Profile CallFactorProfile `json:"profile, omitempty"`
+}
+
+func (m *CallFactor) WithProfile(v CallFactorProfile) *CallFactor {
+	m.Profile = v
+	return m
+}
+
+
+type CallFactorProfile struct {
+	PhoneExtension string `json:"phoneExtension, omitempty"`
+	PhoneNumber string `json:"phoneNumber, omitempty"`
+}
+
+func (m *CallFactorProfile) WithPhoneExtension(v string) *CallFactorProfile {
+	m.PhoneExtension = v
+	return m
+}
+func (m *CallFactorProfile) WithPhoneNumber(v string) *CallFactorProfile {
+	m.PhoneNumber = v
+	return m
+}
+
+
+type EmailFactor struct {
+	Profile EmailFactorProfile `json:"profile, omitempty"`
+}
+
+func (m *EmailFactor) WithProfile(v EmailFactorProfile) *EmailFactor {
+	m.Profile = v
+	return m
+}
+
+
+type EmailFactorProfile struct {
+	Email string `json:"email, omitempty"`
+}
+
+func (m *EmailFactorProfile) WithEmail(v string) *EmailFactorProfile {
+	m.Email = v
+	return m
+}
+
+type FactorResource resource
+
+type Factor struct {
+	Embedded map[string]interface{} `json:"_embedded, omitempty"`
+	Links map[string]interface{} `json:"_links, omitempty"`
+	Device string `json:"device, omitempty"`
+	DeviceType string `json:"deviceType, omitempty"`
+	FactorType string `json:"factorType, omitempty"`
+	Id string `json:"id, omitempty"`
+	MfaStateTokenId string `json:"mfaStateTokenId, omitempty"`
+	Profile FactorProfile `json:"profile, omitempty"`
+	Provider string `json:"provider, omitempty"`
```

Comment:
Shouldn't be a type called `FactorProvider` for `Provider` defined?

Reply:
object model is undefined for provider in the spec